include/boost/multiprecision/cpp_dec_float.hpp

```diff
@@ -3091,37 +3236,73 @@ template <unsigned Digits10, class ExponentType, class Allocator>
 inline void eval_floor(cpp_dec_float<Digits10, ExponentType, Allocator>& result, const cpp_dec_float<Digits10, ExponentType, Allocator>& x)
 {
    result = x;
-   if (!(x.isfinite)() || x.isint())
+
+   if((x.isfinite)() == false)
    {
       if ((x.isnan)())
          errno = EDOM;
-      return;
    }
+   else
+   {
+      result.eval_round_self();
 
-   if (x.isneg())
-      result -= cpp_dec_float<Digits10, ExponentType, Allocator>::one();
-   result = result.extract_integer_part();
+      if(result.isint())
+      {
+         ;
```

Comment:
?

Reply:
> `if(result.isint())`
> ?

Indeed. These sequences have been simlified in recent push(es).